CMakeLists.txt

```diff
@@ -18,10 +18,19 @@ install(FILES
 
 OPTION(BUILD_ACCURACY  "fpm accuracy"  ON)
 OPTION(BUILD_BENCHMARK "fpm benchmark" ON)
+OPTION(BUILD_TESTS     "fpm tests"     ON)
+
+# only build tests & benchmarks if a top-level project
+if (NOT CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+  set(BUILD_ACCURACY  OFF)
+  set(BUILD_BENCHMARK OFF)
+  set(BUILD_TESTS     OFF)
+endif()
```

Comment:
this may or may not be controversial, since it changes existing behavior.

If this needs to go, that's fine, at least with CPM, because the following can be used to include this project without those options set:

```
CPMAddPackage(
    NAME fpm
    GITHUB_REPOSITORY flaviut/fpm
    VERSION 1.1.1
    GIT_TAG 7d05e2001a0b42ebfb67f5004191b1ffd19f662c
    OPTIONS "BUILD_ACCURACY OFF" "BUILD_BENCHMARK OFF" "BUILD_TESTS OFF"
)

Reply:
It does change existing behavior, but after looking around it seems this is the "modern" way to toggle tests, etc. It's hard to imagine a case where someone includes `fpm` in their project but still wants the validation targets, so I'm fine with this change. Thanks!